rs/sns/testing/src/sns.rs

```diff
@@ -109,83 +110,164 @@ pub async fn create_sns<C: CallCanisters + ProgressNetwork>(
+        .filter_map(|(name, sns)| {
+            if name
+                .as_ref()
+                .map(|n| n == &Some(sns_name.clone()))
+                .unwrap_or_default()
+            {
+                Some(sns.clone())
+            } else {
+                None
+            }
+        })
+        .collect()
+}
 
-    let sns_nervous_system_parameters = sns
-        .governance
-        .get_nervous_system_parameters(dev_participant_agent)
+// helper to get the current participation of the given swap participant
+async fn get_current_participation<C: CallCanisters>(
+    agent: &C,
+    swap_canister: SwapCanister,
+) -> u64 {
+    let buyer_state = swap_canister
+        .get_buyer_state(agent, agent.caller().unwrap().into())
         .await
-        .unwrap();
-
-    for swap_participant_agent in swap_participants_agents[0..mininum_participants].iter() {
-        let swap_participant_neuron_id = get_caller_neuron(swap_participant_agent, sns_governance)
-            .await
-            .expect("Failed to get the caller neuron");
-        let follow = Follow {
-            followees: vec![dev_participant_neuron_id.clone()],
-            // UpgradeSnsControlledCanister
-            function_id: 3,
-        };
-        sns_governance
-            .follow(
-                swap_participant_agent,
-                swap_participant_neuron_id.clone(),
-                follow,
-            )
-            .await
-            .expect("Failed to follow the dev neuron");
-
-        sns_governance
-            .increase_dissolve_delay(
-                swap_participant_agent,
-                swap_participant_neuron_id,
-                sns_nervous_system_parameters
-                    .neuron_minimum_dissolve_delay_to_vote_seconds
-                    .unwrap() as u32,
-            )
-            .await
-            .unwrap();
+        .map(|r| r.buyer_state);
+    match buyer_state {
+        Ok(Some(BuyerState {
+            icp: Some(TransferableAmount { amount_e8s, .. }),
+            ..
+        })) => amount_e8s,
+        _ => 0u64,
     }
-
-    sns
 }
 
 // Completes the swap by transferring the required amount of ICP from the "treasury" account
 // and participating in the swap for each participant using agents provided as arguments:
-// 1) swap_treasury_agent - Agent for the identity that has sufficient amout of ICP tokens to close the swap.
-// 2) swap_participants_agents - Agents for the identities that will participate in the swap.
-// 3) swap_parameters - Swap parameters that define the swap.
-// 4) swap_canister - SNS Swap canister on which the swap will be completed.
-async fn complete_sns_swap<C: CallCanisters + ProgressNetwork>(
-    swap_treasury_agent: &C,
-    swap_participants_agents: &[C],
-    swap_parameters: SwapParameters,
+// 1) agent - Agent that is used to provide IC network settings.
+// 2) use_ephemeral_icp_treasury - defines whether the identity of the 'agent' or 'TREASURY_PRINCIPAL_ID'.
```

Comment:
It's unclear from this description how the flag works. How about 

```suggestion
// 2) sponsor_participants_from_agent - defines whether the identity of the 'agent' or 'TREASURY_PRINCIPAL_ID'.
```